internal/handler/users_handler.go

```diff
@@ -79,3 +82,78 @@ func GetUsersHandler(store *db.Store) gin.HandlerFunc {
 		ctx.JSON(http.StatusOK, user)
 	}
 }
+
+// ListUsers handler
+
+type listUsersRequest struct {
+	PageID   int32 `form:"page_id" binding:"required,min=1"`
+	PageSize int32 `form:"page_size" binding:"required,min=5,max=10"`
+}
+
+func ListUsersHandler(store *db.Store) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req listUsersRequest
+		if err := ctx.ShouldBindQuery(&req); err != nil {
+			ctx.JSON(http.StatusBadRequest, errorResponse(err))
+			return
+		}
+
+		args := db.ListUsersParams{
+			Limit:  req.PageSize,
+			Offset: (req.PageID - 1) * req.PageSize,
+		}
+
+		users, err := store.ListUsers(ctx, args)
+		if err != nil {
+			ctx.JSON(http.StatusInternalServerError, errorResponse(err))
+			return
+		}
+		ctx.JSON(http.StatusOK, users)
+	}
+}
+
+// UpdateUsers handler
+
+type updateUsersRequest struct {
+	ID          int64  `json:"id" binding:"required"`
+	Username    string `json:"username" binding:"required"`
+	Email       string `json:"email" binding:"required"`
+	Password    string `json:"password" binding:"required"`
+	Role        string `json:"role" binding:"required"`
+	FirstName   string `json:"first_name" binding:"required"`
+	LastName    string `json:"last_name" binding:"required"`
+	UserUrl     string `json:"user_url" binding:"required"`
+	Description string `json:"description" binding:"required"`
+	IsDeleted   bool   `json:"is_deleted"`
+}
+
+func UpdateUsersHandler(store *db.Store) gin.HandlerFunc {
+	return func(ctx *gin.Context) {
+		var req updateUsersRequest
+		if err := ctx.ShouldBindJSON(&req); err != nil {
+			ctx.JSON(http.StatusBadRequest, errorResponse(err))
+			return
+		}
+
+		args := db.UpdateUsersParams{
+			ID:          req.ID,
```

Comment:
Here you are directly applying `req.id` in order to update a specific user.
Can you guarantee that the client side will provide a `users.id` that is valid and available in the db layer?

Reply:
upadted